dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/assertj/SpanEventAssert.kt

```diff
@@ -0,0 +1,279 @@
+            )
+            .isEqualTo(spanSource)
+        return this
+    }
+
+    fun hasSpanStartTime(startTime: Long): SpanEventAssert {
+        assertThat(actual.start)
+            .overridingErrorMessage(
+                "Expected SpanEvent to have start time: $startTime" +
+                    " but instead was: ${actual.start}"
+            )
+            .isEqualTo(
+                startTime
+            )
+        return this
+    }
+
+    fun hasSpanDuration(duration: Long): SpanEventAssert {
+        assertThat(actual.duration)
+            .overridingErrorMessage(
+                "Expected SpanEvent to have duration: $duration" +
+                    " but instead was: ${actual.duration}"
+            )
+            .isEqualTo(
+                duration
+            )
+        return this
+    }
+
+    fun hasTracerVersion(version: String): SpanEventAssert {
+        assertThat(actual.meta.tracer.version)
+            .overridingErrorMessage(
+                "Expected SpanEvent to have tracer version: $version" +
+                    " but instead was: ${actual.meta.tracer.version}"
+            )
+            .isEqualTo(version)
+        return this
+    }
+
+    fun hasClientPackageVersion(clientPackageVersion: String): SpanEventAssert {
+        assertThat(actual.meta.version)
+            .overridingErrorMessage(
+                "Expected SpanEvent to have client package" +
+                    " version: $clientPackageVersion" +
+                    " but instead was: ${actual.meta.version}"
+            )
+            .isEqualTo(clientPackageVersion)
+        return this
+    }
+
+    fun hasErrorFlag(error: Long): SpanEventAssert {
+        assertThat(actual.error)
+            .overridingErrorMessage(
+                "Expected SpanEvent to have errorFlag: $error" +
+                    " but instead was: ${actual.error}"
+            )
+            .isEqualTo(error)
+        return this
+    }
+
+    fun hasMeta(attributes: Map<String, String>): SpanEventAssert {
+        assertThat(actual.meta.additionalProperties)
+            .hasSameSizeAs(attributes)
+            .containsAllEntriesOf(attributes)
+        return this
+    }
+
+    fun hasMetrics(metrics: Map<String, Number>): SpanEventAssert {
+        assertThat(actual.metrics.additionalProperties)
+            .hasSameSizeAs(metrics)
+            .containsAllEntriesOf(metrics)
+        return this
+    }
+
+    fun isTopSpan(): SpanEventAssert {
+        assertThat(actual.metrics.topLevel).overridingErrorMessage(
+            "Expected the " +
+                " metrics top level to be 1 but instead was ${actual.metrics.topLevel}"
+        ).isEqualTo(1L)
+        return this
+    }
+
+    fun isNotTopSpan(): SpanEventAssert {
+        assertThat(actual.metrics.topLevel).overridingErrorMessage(
+            "Expected the " +
+                " metrics top level to be 0 but instead was ${actual.metrics.topLevel}"
+        ).isEqualTo(1L)
```

Comment:
should be `0L` it seems?

Reply:
Actually it's ok `1L` is just that the error message is wrong.

Reply:
but how it is different from `isTopSpan` then?

Reply:
Oh...wait there's a problem there...I will fix it now.  Nice catch ;)